drivers/timer/Kconfig.nrf_xrtc

```diff
@@ -0,0 +1,42 @@
+# Common RTC configuration
+
+# Copyright (c) 2023 Nordic Semiconductor ASA
+# SPDX-License-Identifier: Apache-2.0
+
+config SYSTEM_CLOCK_WAIT
```

Comment:
Not addressed and agree with @tejlmand why are we breaking users for no reason?

Reply:
It turns out that this extra layer was added, sot that the choice below` was not available for NRF54H. However, as @tejlmand noticed, `SYSTEM_CLOCK_WAIT` was enabled when RTC or GRTC was enabled, so it would probably be enabled for that platform too.
I removed `SYSTEM_CLOCK_WAIT` symbol and changed the if below to `if NRF_RTC_TIMER || NRF_GRTC_TIMER`